app/models/piece.rb

```diff
@@ -1,14 +1,22 @@
+# rubocop:disable Metrics/ClassLength
 class Piece < ActiveRecord::Base
   belongs_to :user
   belongs_to :game
 
   # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/LineLength, Metrics/MethodLength
   def move_to!(new_row, new_col)
-    @piece = Piece.find_by(row_position: new_row, col_position: new_col)
+    Piece.transaction do
+      try_to_move(new_row, new_col)
+      fail ActiveRecord::Rollback if game.determine_check
```

Comment:
Yes, I didn't take care of that yet. If the game is in check we should notify the user "you can't move into check". Maybe move_to! should return true/false, and then in the controller, after calling move_to! we can use an alert message if move_to! returned false. What do you think?